### PR TITLE
Properly camel-case names with dashes

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: codegen/src/main/scala/akka/grpc/gen/javadsl/Service.scala [hunk 1]
```diff
@@ -60,6 +61,20 @@ object Service {
 
   private[javadsl] def toCamelCase(name: String): String = {
     if (name.isEmpty) ""
-    else name.head.toUpper + "_[a-z]".r.replaceAllIn(name.tail, s => s.group(0)(1).toUpper.toString)
+    else toCamelCaseRec(name.tail, new StringBuffer(name.head.toUpper.toString), false)
+  }
+
+  @tailrec
+  def toCamelCaseRec(in: String, out: StringBuffer, capNext: Boolean): String = {
```
Comment (ennru) [inline codegen/src/main/scala/akka/grpc/gen/javadsl/Service.scala]: `private`

File: codegen/src/test/scala/akka/grpc/gen/javadsl/ServiceSpec.scala [hunk 2]
```diff
@@ -11,6 +11,7 @@ class ServiceSpec extends AnyWordSpec with Matchers {
   "The Service model" should {
     "correctly camelcase strings" in {
       Service.toCamelCase("foo_bar") should be("FooBar")
+      Service.toCamelCase("grpc-example") should be("GrpcExample")
```
Comment (ennru) [inline codegen/src/test/scala/akka/grpc/gen/javadsl/ServiceSpec.scala]: Add a test with digits, please.

File: interop-tests/src/main/protobuf/crdt-example.proto [hunk 3]
```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
```
Comment (ennru) [inline interop-tests/src/main/protobuf/crdt-example.proto]: Why is this file in this PR?

Comment (raboof) [reply interop-tests/src/main/protobuf/crdt-example.proto]: Without the change in this PR, this .proto would generate gRPC javadsl service code that referred to a `Crdt-example` outer class, which wouldn't compile, reproducing the original problem.

**Pull request merged**